docs/markdown/Reference-manual.md

```diff
@@ -837,6 +837,25 @@ Then you can use it in `bar2` like this:
 ```
 
 Meson will then do the right thing.
+Meson uses the file extension to guess the programming language.
+Because .C is sometimes used for 'c' code and sometimes for 'cpp'
+code, you must specify the `language: kwarg` like that:
+```meson
+    files('fileA.C', language: 'cpp')
+    files('fileB.C', language: 'c')
+```
+Writing `files('fileB.C')` defaults to 'c', but doing this is
+deprecated, because in the future it might default to 'cpp', breaking
+your code.
```

Comment:
No it won't, don't make predictions about meson development policy.

Reply:
Ok, I can remove that.